rgdc/rgdc/rgdc.py

```diff
@@ -0,0 +1,248 @@
+        acquired: Optional[DATETIME_OR_STR_TUPLE] = None,
+        created: Optional[DATETIME_OR_STR_TUPLE] = None,
+        modified: Optional[DATETIME_OR_STR_TUPLE] = None,
+        datatype: Optional[SEARCH_DATATYPE_CHOICE] = None,
+        instrumentation: Optional[str] = None,
+        num_bands: Optional[Tuple[int, int]] = None,
+        resolution: Optional[Tuple[int, int]] = None,
+        cloud_cover: Optional[Tuple[float, float]] = None,
+        frame_rate: Optional[Tuple[int, int]] = None,
+        limit: Optional[int] = None,
+        offset: Optional[int] = None,
+    ) -> Iterator[Dict]:
+        """
+        Search for geospatial entries based on various criteria.
+
+        For Ranges (Tuples), an entry of `None` means that side of the range is unbounded.
+        E.g. a range of (2, None) is 2 or more, (None, 5) is at most 5, (2, 5) is between 2 and 5.
+
+        Args:
+            query: Either a WKT GeoJSON representation, a GeoJSON string, or a GeoJSON dict.
+            predicate: A named spatial predicate based on the DE-9IM. This spatial predicate will
+                be used to filter data such that predicate(a, b) where b is the queried geometry.
+            relates: Specify exactly how the queried geometry should relate to the data using a
+                DE-9IM string code.
+            distance: The min/max distance around the queried geometry in meters.
+            acquired: The min/max date and time (ISO 8601) when data was acquired.
+            created: The min/max date and time (ISO 8601) when data was created.
+            modified: The min/max date and time (ISO 8601) when data was modified.
+            datatype: The datatype to provide.
+            instrumentation: The instrumentation used to acquire at least one of these data.
+            num_bands: The min/max number of bands in the raster.
+            resolution: The min/max resolution of the raster.
+            cloud_cover: The min/max cloud coverage of the raster.
+            frame_rate: The min/max frame rate of the video.
+            limit: The maximum number of results to return.
+            offset: The number of results to skip.
+
+        Returns:
+            An iterator of Spatial Entries.
+        """
+        # The dict that will be used to store params.
+        # Initialize with queries that won't be additionally processed.
+        params = {
+            'predicate': predicate,
+            'relates': relates,
+            'datatype': datatype,
+            'instrumentation': instrumentation,
+            'limit': limit,
+            'offset': offset,
+        }
+
+        if query:
+            if isinstance(query, str):
+                try:
+                    query = json.loads(query)
+                except JSONDecodeError:
+                    pass
+
+            if isinstance(query, dict):
+                # Allow failure on invalid format
+                query = wkt.dumps()
```

Comment:
is this supposed to be?:

```suggestion
                query = wkt.dumps(query)
```